src/compose/volume-manager.ts

```diff
@@ -23,9 +24,29 @@ export async function get({ name, appId }: VolumeNameOpts): Promise<Volume> {
 	);
 }
 
-export async function getAll(): Promise<Volume[]> {
-	const volumeInspect = await listWithBothLabels();
-	return volumeInspect.map((inspect) => Volume.fromDockerVolume(inspect));
+export async function getAll(allowLegacy = false): Promise<Volume[]> {
+	let volumes;
+	if (allowLegacy) {
+		// Returns all Volumes which might not have a supervised label
+		volumes = await list();
+	} else {
+		// Only get Volumes with supervised label
+		volumes = await listByLabel();
+	}
+	// Normalize inspect information to Volume types and filter any that fail
+	return volumes.reduce((volumesList, volumeInfo) => {
+		try {
+			const volume = Volume.fromDockerVolume(volumeInfo);
+			volumesList.push(volume);
+		} catch (e) {
+			if (e.message.substring(0, 16) === 'Could not detect') {
```

Comment:
I think creating a new exception type and throw that in `Volume.deconstructDockerName` might be  a better option here @20k-ultra

Reply:
I think this doesn't work for correctly named volumes without the `io.balena.supervised` label, right? In that case `Volume.deconstructDockerName` doesn't throw and it is also not catched by `listByLabel`

Reply:
Is there any drawback to always using `list()` instead of `listByLabel`? If there isn't I would get rid of the function parameter and just use `list()` by default

Reply:
I agree with creating a new exception type that we can use for better error handling of this type. I'll do this.

This error is only for failure to parse a volume name, there's no need for it to be aware of the `io.balena.supervised` label so your observations are by design.

I kept listByLabel (previously listWithBothLabels) even though no one is using it because when I'm patching an issue I want to avoid introducing new regressions at all costs so I write the patch in a way that fixes the issue without introducing new behavior. By using this toggle (allowLegacy) the codebase acts the same unless when this value is true. However, whenever I make a PR patch and the reviewer thinks there's optimizations we can squeeze in at the same time I'm cool to add them. i'll clean this up too.

Reply:
Oh, I thought those volumes without the label were causing the issue. I see now.

For  the adding the `allowLegacy` arg, if you feel it might be safer to do it that way, then let's keep it. It is true that just relying on naming to identify volumes might be easy to break, so do what you believe it's best